src/main/scala/org/dbpedia/spotlight/WikipediaParser.scala

```diff
@@ -0,0 +1,128 @@
+/**
+ *  Copyright 2015 DBpedia Spotlight
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+package org.dbpedia.spotlight
+
+import com.sun.xml.internal.bind.v2.TODO
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.dbpedia.util.DBpediaUriEncode
+import org.dbpedia.wiki.format.XmlInputFormat
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.io.{LongWritable, Text}
+import org.apache.spark.SparkContext
+import org.apache.spark.sql
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.JavaConversions._
+
+
+class WikipediaParser(inputWikiDump:String,lang:String)(implicit val sc: SparkContext){
```

Comment:
can we rename this to `JsonpediaParser` and make it such that it only contains the code that knows how to parse the jsonpedia i.e: reading the dataframes etc..